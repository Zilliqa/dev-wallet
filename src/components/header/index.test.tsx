/**
 * This file is part of nucleus-wallet.
 * Copyright (c) 2018 - present Zilliqa Research Pte. Ltd.
 *
 * nucleus-wallet is free software: you can redistribute it and/or modify it under the
 * terms of the GNU General Public License as published by the Free Software
 * Foundation, either version 3 of the License, or (at your option) any later
 * version.
 *
 * nucleus-wallet is distributed in the hope that it will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along with
 * nucleus-wallet.  If not, see <http://www.gnu.org/licenses/>.
 */

import React from 'react';
import { render } from '@testing-library/react';
import Header from '.';
import { MemoryRouter } from 'react-router';

test('matches the snapshot', () => {
  const clearAuth = jest.fn();
  const isAuth = false;
  const { container } = render(
    <MemoryRouter>
      <Header isAuth={isAuth} clearAuth={clearAuth} />
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});

test('matches the snapshot', () => {
  const clearAuth = jest.fn();
  const isAuth = true;
  const { container } = render(
    <MemoryRouter>
      <Header isAuth={isAuth} clearAuth={clearAuth} />
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});
