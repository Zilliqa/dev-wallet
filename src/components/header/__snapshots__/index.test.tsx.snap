// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div>
  <div>
    <nav
      class="navbar navbar-expand-sm navbar-dark bg-dark fixed-top"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        Nucleus Wallet
        <small
          class="release-text"
        >
          beta
        </small>
      </a>
      <button
        aria-label="toggler"
        class="navbar-toggler"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
      >
        <ul
          class="ml-auto navbar-nav"
        >
          <li
            class="sidebar-link nav-item"
          >
            <a
              class="nav-link"
              href="/home"
            >
              Home
            </a>
          </li>
          <li
            class="sidebar-link nav-item"
          >
            <a
              class="nav-link"
              href="/generate"
            >
              Create New Wallet
            </a>
          </li>
          <li
            class="sidebar-link nav-item"
          >
            <a
              class="nav-link"
              href="/send"
            >
              Send ZIL
            </a>
          </li>
          <li
            class="sidebar-link nav-item"
          >
            <a
              class="nav-link"
              href="/faucet"
            >
              ZIL Faucet
            </a>
          </li>
          <span
            class="nav-link"
          >
            <span
              class="network"
            >
              Dev Testnet
            </span>
          </span>
        </ul>
      </div>
    </nav>
  </div>
</div>
`;

exports[`matches the snapshot 2`] = `
<div>
  <div>
    <nav
      class="navbar navbar-expand-sm navbar-dark bg-dark fixed-top"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        Nucleus Wallet
        <small
          class="release-text"
        >
          beta
        </small>
      </a>
      <button
        aria-label="toggler"
        class="navbar-toggler"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
      >
        <ul
          class="ml-auto navbar-nav"
        >
          <li
            class="sidebar-link nav-item"
          >
            <a
              class="nav-link"
              href="/home"
            >
              Home
            </a>
          </li>
          <li
            class="sidebar-link nav-item"
          >
            <a
              class="nav-link"
              href="/generate"
            >
              Create New Wallet
            </a>
          </li>
          <li
            class="sidebar-link nav-item"
          >
            <a
              class="nav-link"
              href="/send"
            >
              Send ZIL
            </a>
          </li>
          <li
            class="sidebar-link nav-item"
          >
            <a
              class="nav-link"
              href="/faucet"
            >
              ZIL Faucet
            </a>
          </li>
          <a
            class="cursor-pointer nav-link"
          >
            Sign Out
          </a>
          <span
            class="nav-link"
          >
            <span
              class="network"
            >
              Dev Testnet
            </span>
          </span>
        </ul>
      </div>
    </nav>
  </div>
</div>
`;
